cylc/flow/parsec/fileparse.py

```diff
@@ -224,6 +228,31 @@ def read_and_proc(fpath, template_vars=None, viewcfg=None, asedit=False):
     do_empy = True
     do_jinja2 = True
     do_contin = True
+
+    if not template_vars:
+        template_vars = {}
+
+    # Load Rose Vars, if a ``rose-suite.conf`` file is present.
+    extra_vars = {
+        'env': {},
+        'template variables': None,
+        'templating detected': None
+    }
```

Comment:
Naming could be tweaked a bit, since this is essentially a public interface (no spaces, so you can use the names as kwargs `extra_vars.update(template_variables={})`:
```suggestion
    extra_vars = {
        'environment_variables': {},
        'template_variables': None,
        'template_engine': None
    }
```

Reply:
Good idea.